enhancements/network/multi-network-vrfs.md

```diff
@@ -0,0 +1,499 @@
+
+Furthermore, the internal masquerade subnet and the Kubernetes service subnet will remain unique and will exist globally to serve all networks. The join subnets and transit switch subnets will overlap between networks, and are not exposed outside of the OVN topology.
+
+
+### DNS
+
+DNS lookups will happen via every pod’s access to the DNS service on the default network. CoreDNS lookups for pods will resolve to the pod’s IP on the default network. This is a limitation of the first phase of this feature and will be addressed in a future enhancement. DNS lookups for services and external entities will function correctly.
+
+
+### Services
+
+Services in Kubernetes are namespace scoped. Any creation of a service in a namespace without a unique network (using default network as primary) will only be accessible by other namespaces also using the default network as their primary network. Services created in namespaces served by unique networks, will only be accessible to namespaces connected to the unique network. Two caveats to this are:
+
+
+
+1. Host networked pods of any namespace are able to reach any service.
+2. Pods that are connected to unique networks are still able to access KAPI and DNS services that reside on the default network.
+
+With this proposal, OVN-Kubernetes will check which network is being used for this namespace, and then only enable the service there. The cluster IP of the service will only be available in the network of that service. The one caveat to this is that the service will still be reachable from any host via Cluster IP. Load balancer IP and nodeport services are also supported on unique networks. Service selectors will only be able to select endpoints for the namespace(s) attached to the network. Services that exist before the unique network is assigned to a namespace will result in OVN-Kubernetes executing a re-sync on all services in that namespace, and updating all load balancers. Keep in mind that, pods may not exist in the namespace when the namespace is assigned to a new network or the new network assignment will not be accepted by OVN-Kubernetes.
+
+When multiple namespaces share the same network, services will be reachable in between namespaces. It is also possible to select endpoints between both namespaces for these services.
+
+As previously mentioned, Kubernetes API and DNS services will be accessible by all pods.
+
+Endpoint slices will provide the IPs of the default network in Kubernetes API. For this implementation the required endpoints are those IP addresses which reside on the unique primary network. In order to solve this problem, OVN-Kubernetes may create its own endpoint slices or may choose to do dynamic lookups at runtime to map endpoints to their primary IP address. Leveraging a second set of endpoint slices will be the preferred method, as it creates less indirection and gives explicit Kube API access to what IP addresses are being used by OVN-Kubernetes.
```

Comment:
The endpoints used internally should be rendered somehow, either as annotations on the epslice or as another epslice, otherwise debugging / triaging may become hard. Also, please note that other controllers (namely, MetalLB) may rely on the presence of endpoint slices to check the status and understand if a node should serve the service (in case of local etp).
As of today, MetalLB only check for the presence of _at least one_ healthy endpoint, so in case of cloning we can just rely on the one produced by the "official" ep slices controller.

Reply:
exactly my thought